libcloudforensics/providers/azure/internal/account.py

```diff
@@ -0,0 +1,221 @@
+
+
+class AZAccount:
+  """Class that represents an Azure Account.
+
+  Attributes:
+    subscription_id (str): The Azure subscription ID to use.
+    credentials (ServicePrincipalCredentials): An Azure credentials object.
+    compute_client (ComputeManagementClient): An Azure compute client object.
+  """
+
+  def __init__(self, subscription_id: str) -> None:
+    """Initialize the AZAccount class.
+
+    Args:
+      subscription_id (str): The Azure subscription ID to use.
+    """
+    self.subscription_id = subscription_id
+    self.credentials = common.GetCredentials()
+    self.compute_client = ComputeManagementClient(
+        self.credentials, self.subscription_id)
+
+  def ListInstances(
+      self,
+      # pylint: disable=line-too-long
+      resource_group_name: Optional[str] = None) -> Dict[str, compute.AZVirtualMachine]:
+      # pylint: enable=line-too-long
+    """List instances in an Azure subscription / resource group.
+
+    Args:
+      resource_group_name (str): Optional. The resource group name to list
+          instances from. If none specified, then all instances in the AZ
+          subscription will be listed.
+
+    Returns:
+      Dict[str, AZCompute]: Dictionary mapping instance names (str) to their
+          respective AZCompute object.
+    """
+    instances = {}  # type: Dict[str, compute.AZVirtualMachine]
+    az_vm_client = self.compute_client.virtual_machines
+    if not resource_group_name:
+      responses = common.ExecuteRequest(az_vm_client, 'list_all')
+    else:
+      responses = common.ExecuteRequest(
+          az_vm_client,
+          'list',
+          {'resource_group_name': resource_group_name})  # type: ignore
+    for response in responses:
+      for instance in response:
+        vm_name = instance.name
+        instances[vm_name] = compute.AZVirtualMachine(self,
+                                                      instance.id,
+                                                      vm_name,
+                                                      instance.location,
+                                                      zones=instance.zones)
+    return instances
+
+  def ListDisks(
+      self,
+      # pylint: disable=line-too-long
+      resource_group_name: Optional[str] = None) -> Dict[str, compute.AZDisk]:
+      # pylint: enable=line-too-long
+    """List disks in an Azure subscription / resource group.
+
+    Args:
+      resource_group_name (str): Optional. The resource group name to list
+          disks from. If none specified, then all disks in the AZ
+          subscription will be listed.
+
+    Returns:
+      Dict[str, AZDisk]: Dictionary mapping disk names (str) to their
+          respective AZDisk object.
+    """
+    disks = {}  # type: Dict[str, compute.AZDisk]
+    az_disk_client = self.compute_client.disks
+    if not resource_group_name:
+      responses = common.ExecuteRequest(az_disk_client, 'list')
+    else:
```

Comment:
**kwargs??

Reply:
The functions called are different (not only the parameters), not sure how I can use **kwargs there

Reply:
You seem to be calling `common.ExecuteRequest` both times, but I agree it's clearer to have two separate calls since the parameters are very different.